src/main/java/players/mcts/SingleTreeNode.java

```diff
@@ -856,11 +860,13 @@ protected double[] rollout(double[] startingValues, int lastActor) {
                 root.copyCount++;
             }
 
+            AbstractAction next = null;
             while (!finishRollout(rolloutState)) {
-                List<AbstractAction> availableActions = forwardModel.computeAvailableActions(rolloutState);
-                if (availableActions.isEmpty())
-                    throw new AssertionError("No actions available in rollout!");
-                AbstractAction next = opponentModels[rolloutState.getCurrentPlayer()].getAction(rolloutState, availableActions);
+                List<AbstractAction> availableActions = forwardModel.computeAvailableActions(rolloutState, params.actionSpace);
+                if (availableActions.isEmpty()) {
+                    throw new AssertionError("No actions available in rollout!" + (next != null? " Last action: " + next.toString() : ""));
+                }
+                next = opponentModels[rolloutState.getCurrentPlayer()].getAction(rolloutState);
```

Comment:
As previous comment - we are duplicating effort by calculating all actions twice.